### PR TITLE
Change example of computed property example to avoid shared state confusion

### LINKED ISSUE #1699: Computed properties example uses unexpected shared state in component
Hi,

Looking at the example on https://github.com/emberjs/guides/blob/master/source/localizable/object-model/computed-properties-and-aggregate-data.md:

```
export default Ember.Component.extend({
  todos: [
    Ember.Object.create({ isDone: true }),
    Ember.Object.create({ isDone: false }),
    Ember.Object.create({ isDone: true })
  ],

  incomplete: Ember.computed('todos.@each.isDone', function() {
    var todos = this.get('todos');
    return todos.filterBy('isDone', false);
  })
});
```

is not clear that the value of `todos` will be shared across all instances of the component in the app. Even if the component is only used once in the application, multiple running tests will share that state and produce undesired effects.

It might be worth thinking if we could modify the example to be:

```
export default Ember.Component.extend({
  todos: Ember.computed(function() {
    return [
      Ember.Object.create({ isDone: true }),
      Ember.Object.create({ isDone: false }),
      Ember.Object.create({ isDone: true })
    ];
  }),

  incomplete: Ember.computed('todos.@each.isDone', function() {
    var todos = this.get('todos');
    return todos.filterBy('isDone', false);
  })
});
```

To avoid the side effects mentioned above.

### PR DESCRIPTION
**Pull request by juanazam**
Closes #1699

### CODE REVIEW

File: source/localizable/object-model/computed-properties-and-aggregate-data.md [hunk 1]
```diff
@@ -6,11 +6,13 @@ To facilitate this, Ember provides the `@each` key illustrated below:
 
 ```app/components/todo-list.js
 export default Ember.Component.extend({
-  todos: [
-    Ember.Object.create({ isDone: true }),
-    Ember.Object.create({ isDone: false }),
-    Ember.Object.create({ isDone: true })
-  ],
+  todos: Ember.computed(function() {
```
Comment (locks) [inline source/localizable/object-model/computed-properties-and-aggregate-data.md]: It is preferable to initialize this sort of data in an init, like so:

```
export default Ember.Component.extend({
  todos: null,

  init() {
    this.set('todos', [
      Ember.Object.create({ isDone: true }),
      Ember.Object.create({ isDone: false }),
      Ember.Object.create({ isDone: true }),
    ]);
  }
});
```

Comment (juanazam) [inline source/localizable/object-model/computed-properties-and-aggregate-data.md]: 👍

Comment (juanazam) [inline source/localizable/object-model/computed-properties-and-aggregate-data.md]: Done!

**Comment by locks** [comment]
Thanks for your contribution!

@homu r+

**Pull request merged**